presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVerifier.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+public interface PlanVerifier
```

Comment:
No. PlanSanityChecker is called in the middle of logical plan generation (e.g. between optimizers). While PlanVerifiers are run only at the end of logical plan generation.